pkg/controller/node/controller_utils.go

```diff
@@ -110,6 +111,10 @@ func deletePods(kubeClient clientset.Interface, recorder record.EventRecorder, n
 		if err == nil { // No error means at least one daemonset was found
 			continue
 		}
+		// if the pod is forgivable (wants to stay bound on the node), ignore it
+		if isPodForgivable(pod, nodeStatus.readyTransitionTimestamp) {
```

Comment:
This whole code path really needs an in depth test - I agree with @gmarek that given the history of issues we've had that without a test we can't merge this.